cpp/src/gandiva/expression_registry.h

```diff
@@ -61,5 +61,9 @@ class ExpressionRegistry {
   static void AddArrowTypesToVector(arrow::Type::type& type, DataTypeVector& vector);
   std::unique_ptr<FunctionRegistry> function_registry_;
 };
+
+void GetRegisteredFunctionSignatures(
```

Comment:
curious - why not 

std::vector<std::shared_ptr<FunctionSignature>>* signatures) GetRegisteredFunctions() ?

Reply:
In general I like the pattern where the client allocates the vector and the function pushes onto it, it allows adding elements to an existing vector. In this case I agree it's not necessary, I changed it to your suggestion.